### ISSUE TITLE
v2 API endpoint returns 404 error

### ISSUE THREAD

**Issue by theremon** [VivaPayments/API#1011]
This issue is related to #944, which should not have been closed as far as I'm concerned.
The documentation needs serious updating. There is a mixup as far as the URLs are concerned - and the PHP demo scripts are obsolete as well.

I'm trying to send a 'create order' request (as we did with v1), but keep getting 404 errors for the new v2 API URLs.
In #944, the 'correct' URL is https://demo.vivapayments.com/api/orders, but that is valid for the v1 API and does not appear to work with v2 API calls since the authentication method has changed.
I'm getting a 404 error as well for https://demo-api.vivapayments.com/orders, https://demo-api.vivapayments.com/api/orders and the respective production URLs as well.
Is there a valid v2 API URL we can work with?
If not, when do you expect to have it available since the 3DS methods should be available in a few days and the v1 API is already deprecated?

**Comment by theremon** [comment]
Pasting reply initially sent to issue #944 by :
> @theremon Yes, from what I've seen you have to use both variants (I don't understand why and assume it's some weirdness with the demo systems).
> 
> Generate a token on your server, this assumes PHP and is using the Guzzle HTTP client for PHP. I believe it should be relatively straight forward to port to other languages though.
> 
> ```
> $client = new \GuzzleHttp\Client();
> $res = $client->request('POST','https://demo-accounts.vivapayments.com/connect/token', [
>         "auth"        => [ 'client_id', 'client_secret' ],
>         "form_params" => [
>             'grant_type' => 'client_credentials',
>         ],
>     ]
> );
> 
> $result = json_decode($res->getBody());
> 
> // validate successful request
> 
> // $result now contains an access_token, this will need to be passed to your JS implementation
> ```
> 
> To get a charge token (this assumes you're using Viva's JS SDK) you setup the library with the `https://demo-api.vivapayments.com` endpoint:
> 
> ```js
> VivaPayments.cards.setup({
>     authToken: <access token generated on the server>,
>     baseURL: 'https://demo-api.vivapayments.com',
>     cardHolderAuthOptions: {
>         cardHolderAuthPlaceholderId: 'viva_native_threed-secure-pane',
>         cardHolderAuthInitiated: function () {
>             threedModalElement.style.display = "block";
>         },
>         cardHolderAuthFinished: function () {
>             setTimeout(function () {
>                 threedModalElement.style.display = "none";
>             }, 1000);
>         }
>     }
> });
> ```
> 
> Now you can perform a charge token request, once the form has been completed.
> 
> ```js
> VivaPayments.cards.requestToken({
>     amount: <amount in smallest value>,
>     authenticateCardholder: true
> }).then(function (response) {
>     // response.chargeToken will be the token to parse to the server
> }, function (error) {
>     // error with the request
> });
> ```
> 
> On your server you can then use the charge token to create an order, and then create a transaction with the order. The documentation for these API requests are correct I believe, but if you're having trouble with this also I can provide an example for both too 😄.
> 
> You use `https://demo.vivapayments.com/api/orders` and `https://demo.vivapayments.com/api/transactions` for that.
> 
> The orders and transactions APIs use the normal bearer auth method, same as when you requested the access token above.
> 
> When you move to production the documentation endpoints are correct I believe:
> 
> * Account - `https://accounts.vivapayments.com/`
> * API - `https://api.vivapayments.com/` (this does apply to both the server API calls, and the JS request token calls this time)
> 
> I have an implementation of the above working on a production e-commerce store.
> 
> I totally agree that the documentation needs a massive overhaul, took me 2 - 3 days ripping my hair out and trial and error to get this working. I was fortunate enough to be getting paid while doing it, so wasn't a 'waste' of my time, but was a waste on my employers' money!
> 
> Hope this helps! I know how frustrating it was for me. But once it's setup, it's been super reliable since.
> 
> **Edit:** Forgot to add, when testing payments use `4016000000040000012` as your card number. This is the only one I've found to perform a successful payment, and provide a reliable 3D Secure popup. 👍

**Comment by matthewrgourd** [comment]
Hi @theremon, I gather my colleague has now replied to the previously closed issue (see below) https://github.com/VivaPayments/API/issues/944#issuecomment-524896905
Please let me know if you have any outstanding queries. As regards the documentation, we want to continue making improvements and you can rest assured that your comments have been taken on board. Thanks!

**Comment by theremon** [comment]
Hello @matthewrgourd 

thanks for the reply. Following what ultimately came to light in #944, it seems that we got a working solution - or at least got past all the problems we faced due to the bad migration instructions and the setup problems.
Now we cannot test the final code because there is no testing card provided that requires 3DS verification. Hopefully Viva will provide one and we'll be able to continue testing.
